src/agent/configuration_parser/include/configuration_parser.hpp

```diff
@@ -100,6 +109,10 @@ namespace configuration
                 {
                     return ParseTimeUnit(current.as<std::string>());
                 }
+                else if constexpr (std::is_same_v<T, std::size_t>)
```

Comment:
Be careful about https://github.com/wazuh/wazuh-agent/blob/master/src/agent/src/agent.cpp#L62:
```cpp
    m_taskManager.Start(
        m_configurationParser->GetConfig<size_t>("agent", "thread_count").value_or(config::DEFAULT_THREAD_COUNT));
```

Reply:
If you want to parse it as a size_t without a unit it will take it as the number itself (without a multiplier).
I don't see any conflict with it.

Reply:
For the above comment I have create this [issue](https://github.com/wazuh/wazuh-agent/issues/419)